### DISCUSSION TITLE
Black screen/ not working

### DISCUSSION THREAD

**Discussion by Gor80hd** [Mikubill/sd-webui-controlnet#381] [Q&A]
I just installed Controlnet and it doesn't work in openpose mode. in others it seems to work, but openpose gives a black screen
![image](https://user-images.githubusercontent.com/92448318/221386244-809150b5-3c9d-4ed4-b8b2-7959bc17bd15.png)
![image](https://user-images.githubusercontent.com/92448318/221386245-e6b25413-736b-4fe3-aeea-1fc49d59f640.png)

**Comment by ljleb** [comment]
set preprocessor to `none` when you want to put your own pose. otherwise the preprocessor tries to find a person in the input and it does not recognize stickmen.

**Reply by Gor80hd** [reply]
This produces very strange results
![image](https://user-images.githubusercontent.com/92448318/221396805-3c829076-6ec0-4a24-9f1a-cc6f8e8ffae4.png)

**Reply by ljleb** [reply]
you have to use the right model. The model you want to use here is probably the openpose model, instead of canny. Keep preprocessor to `none`

**Reply by Gor80hd** [reply]
I understood now. I just downloaded the wrong model from civitai. Is there any guide explaining the Preprocessor? Just as I understand stickerman, this is openpose?

**Reply by ljleb** [reply]
the preprocessors are useful when you want to infer detectmaps from a real image. So for example, in the case of openpose, if you want to infer the pose stick figure from a real image with a person in it, you use the `openpose` preprocessor to convert the image into a stick figure. Otherwise, if you already have a raw stick figure, you dont need to preprocess it to feed it into the controlnet model, so you can set preprocessor to none in this case. Same goes with all other preprocessor/model pair.
